client/verta/verta/_internal_utils/_artifact_utils.py

```diff
@@ -275,10 +281,18 @@ def serialize_model(model):
         elif module_name.startswith("tensorflow.python.keras"):
             model_type = "tensorflow"
             tempf = tempfile.NamedTemporaryFile()
-            if get_tensorflow_major_version() == 2:  # save_format param may not exist in TF 1.X
-                model.save(tempf.name, save_format='h5')  # TF 2.X uses SavedModel by default
-            else:
-                model.save(tempf.name)
+            try:
+                if get_tensorflow_major_version() == 2:  # save_format param may not exist in TF 1.X
+                    model.save(tempf.name, save_format='h5')  # TF 2.X uses SavedModel by default
+                else:
+                    model.save(tempf.name)
+            except TypeError as e:
+                h5py = maybe_dependency("h5py")
+                if (str(e) == "a bytes-like object is required, not 'str'"
```

Comment:
Could you add a note to this string comparison? Seem pretty brittle and misterious.